lib/buffer.js

```diff
@@ -392,17 +392,25 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
-Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+Buffer.prototype.indexOf = function indexOf(val, byteOffset, encoding) {
   if (byteOffset > 0x7fffffff)
     byteOffset = 0x7fffffff;
   else if (byteOffset < -0x80000000)
     byteOffset = -0x80000000;
   byteOffset >>= 0;
 
-  if (typeof val === 'string')
-    return binding.indexOfString(this, val, byteOffset);
+  if (typeof val === 'string') {
+    if (encoding === undefined ||
+        encoding === 'ucs2' ||
+        encoding === 'utf16le' ||
+        encoding === 'utf8') {
+      return binding.indexOfString(this, val, byteOffset, encoding);
+    }
+    return binding.indexOfBuffer(
+        this, Buffer(val, encoding), byteOffset, encoding);
```

Comment:
gah. you're right, but that's going to be a hit. though `String::WrtieOneByte()` will work if the encoding is `binary`. i'll worry about moving the others into C++ in the future.